MI5-88 Dohat sirovih podataka iz vanjskog sustava

kmilos-foi opened the pull request:
Kreirana je funkcionalnost dohvata podataka iz vanjskog sustava. Iz vanjskog se sustava dohvaćaju transakcije, Mid-ovi, Tid-ovi i Merchanti. Podaci se spremaju u bazu podataka kako bi se kasnije mogli slati servisu za upravljanje transakcijama.

kmilos-foi committed:
MI5-88 Kreirana klasa Tid

kmilos-foi committed:
MI5-88 Stvorena klasa Mid

kmilos-foi committed:
MI5-88 Stvorena klasa Merchant

kmilos-foi committed:
MI5-88 Stvorene enum klase

Stvorene su enum klase koje se koriste za transakcije. To su CardBrand, InstallmentsCreditor i TrxType.

kmilos-foi committed:
MI5-88 Stvorena klasa Transaction

kmilos-foi committed:
MI5-88 Ažurirane klase Tid i Mid

Klase Tid i Mid su sada povezane pošto jedan Mid objekt sadrži više Tid objekata. Isto tako, jedan Mid pripada točno jednom Merchantu, a jedan Tid sadrži više transakcija.

kmilos-foi committed:
MI5-88 Omogućeno periodično pozivanje metoda

Pomoću RestTemplatea, moći će se periodično pozivati odeređene metode za dohvat podataka s vanjskog sustava.

kmilos-foi committed:
MI5-88 Kreiran TransactionRepository

kmilos-foi committed:
MI5-88 Kreiran MidRepository

kmilos-foi committed:
MI5-88 Kreiran MerchantRepository

kmilos-foi committed:
MI5-88 Kreiran TidsResponse

TidsResponse je DTO u koji se pohranjuju tid-ovi nakon njihovog dohvata iz vanjskog sustava.

kmilos-foi committed:
MI5-88 Kreiran TidRepository

kmilos-foi committed:
MI5-88 Kreiran MerchantResponse

MerchantResponse je DTO u koji sadrži podatke o Merchantu nakon njegovog dohvata iz vanjskog sustava.

kmilos-foi committed:
MI5-88 Kreiran MidResponse

MidResponse je DTO koji sadrži podatke o Mid-u nakon njegovog dohvata iz vanjskog sustava.

kmilos-foi committed:
MI5-88 Dorađena klasa Mid

Dodani su atributi koji su nedostajali.

kmilos-foi committed:
MI5-88 Ispravljena greška u Tid klasi

Atribut mid spajao se na krivi id, no sada se spaja na ispravan id (pos_mid).

kmilos-foi committed:
MI5-88 Kreiran Scheduler

Schedulerova metoda periodicDataFetch periodično dohvaća podatke iz vanjskog sustava.

kmilos-foi committed:
MI5-88 Kreiran DataFetchService

DataFetchService sadrži metodu fetchData koja se poziva kroz Scheduler. Metoda koristi privatne metode za dohvat tid-ova, mid-ova i merchanta, a zatim dohvaćenim podacima postavlja vanjske ključeve i sprema ih u bazu podataka.

kmilos-foi committed:
MI5-88 Dodana metoda za dohvat merchanta po oib-u

kmilos-foi committed:
MI5-88 Dodan API base URL u properties

Base URL za API kojega pruža 7Pay dodan je u application.properties. Sada više nije potrebno pisati cijeli url u servisu, već se može koristiti varijabla za base url.

kmilos-foi committed:
MI5-88 Refaktorirane metode za dohvat podataka

kmilos-foi committed:
MI5-88 Refaktorirana metoda fetchData

fetchData sada se zove synchronizeData, a dijelovi logike za spremanje podataka sada su prebačeni u zasebne privatne metode (saveMerchant, saveMid, saveTid).

kmilos-foi committed:
MI5-88 Ažuriran TidsResponse

kmilos-foi committed:
MI5-88 Kreiran TransactionsResponse

Ovo je DTO klasa u kojoj su sadržane dohvaćene transakcije iz vanjskog sustava.

kmilos-foi committed:
MI5-88 Ispravljene greške u klasi Transaction

TIp podatka za guid je sada UUID, a maskedPin je preimenovan u maskedPan

kmilos-foi committed:
MI5-88 Ispravljena greška u TransactionRepository

Pošto se promijenio tip identifikatora u UUID, u repozitoriju se morao postaviti UUID umjesto stringa.

kmilos-foi committed:
MI5-88 Dodane metode za dohvat transakcija

kmilos-foi committed:
MI5-88 Refaktoriran Scheaduler

Scheaduler je preimenovan u DataSyncScheduler, a period dohvaćanja podataka postavljen je na svakih 12 sati (popodne i u pola noći).

kmilos-foi committed:
MI5-88 Kreiran SevenPayApiClient

SevenPayApiClient sadrži metode koje komuniciraju s vanjskom komponentom tvrtke 7Pay.

kmilos-foi committed:
MI5-88 Ažuriran Scheduler

kmilos-foi committed:
MI5-88 Kreirane servisne klase

Kreirane su servisne klase za Merchanta, Mid-a, Tid-a i Transactions. U njima se sada nalaze metode koje komuniciraju s repozitorijom, a to  u metode za spremanje odgovarajućih podataka.

kmilos-foi committed:
MI5-88 Promijenjen naziv DataSyncSchedulera

kmilos-foi committed:
MI5-88 Promijenjen naziv DataFetchServicea

DataFetchService sada se zove DataFetchHandler, a nalazi se u folderu handlers.

kmilos-foi committed:
MI5-88 Promijenjen naziv metode periodicDataSync

Razlog zbog promjene naziva metode periodicDataSync u periodicDataFetch jest jer "sync" asocira na sinkronizaciju podataka između 2 servisa, no to nije slučaj ovdje. Ovdje se radi samo o dohvaćanju i spremanju podataka.

kmilos-foi committed:
MI5-88 Dohvaćanje podataka kod pokretanja servera

@PostConstruct omogućava da se ova metoda pozove odmah kada se server pokrene.

kmilos-foi committed:
MI5-88 Dodana notacija JsonProperty na polja klasa

Dodana je notacija zbog veće sigurnosti tijekom serijalizacije podataka.

NinoLoncar commented on src/main/java/foi/air/szokpt/transfetch/handlers/DataFetchHandler.java:
```diff
@@ -0,0 +1,54 @@
+        List<Tid> tids = apiClient.fetchTids();
+        tids.forEach(tid -> {
+            Mid mid = apiClient.fetchMidByTid(tid.getPosTid());
+            if (mid != null) {
```
Da se izbjegne ugnježđavanje selekcije u selekciju može se staviti if(mid==null) continue.

NinoLoncar reviewed (approved):
Sve izgleda u redu.

kmilos-foi committed:
MI5-88 Dodana logika za nepoznate brandove

kmilos-foi committed:
MI5-88 Dodana @Enumerated notacija

@Enumerated notacija kod enum polja služi kako bi se se u bazi podataka odgovarajući podaci spremali kao string, a ne kao broj. Primjerice, u bazi će se spremiti VISA, a ne broj 0 kao card_brand.

kmilos-foi committed:
MI5-88 Refaktorirana metoda fetchData

abernobic21 reviewed (approved):
Sad sve izgleda jako dobro.
